regolith/helpers/l_grantshelper.py

```diff
@@ -0,0 +1,94 @@
+"""Helper for listing upcoming (and past) grants.
+
+"""
+import datetime as dt
+import dateutil.parser as date_parser
+from dateutil.relativedelta import relativedelta
+import sys
+
+from regolith.dates import get_dates, is_current
+from regolith.helpers.basehelper import SoutHelperBase
+from regolith.fsclient import _id_key
+from regolith.tools import (
+    all_docs_from_collection,
+    get_pi_id,
+)
+
+TARGET_COLL = "grants"
+HELPER_TARGET = "l_grants"
+
+
+def subparser(subpi):
+    subpi.add_argument("-d", "--date",
+                       help="Filter grants by a date in ISO format (YYYY-MM-DD)"
+                       )
+    subpi.add_argument("-c", "--current", action="store_true", help='outputs only the current grants')
+    return subpi
+
+
+class GrantsListerHelper(SoutHelperBase):
+    """Helper for listing upcoming (and past) grants.
+
+    """
+    # btype must be the same as helper target in helper.py
+    btype = HELPER_TARGET
+    needed_dbs = [f'{TARGET_COLL}']
+
+    def construct_global_ctx(self):
+        """Constructs the global context"""
+        super().construct_global_ctx()
+        gtx = self.gtx
+        rc = self.rc
+        if "groups" in self.needed_dbs:
+            rc.pi_id = get_pi_id(rc)
+        rc.coll = f"{TARGET_COLL}"
+        try:
+            if not rc.database:
+                rc.database = rc.databases[0]["name"]
+        except:
+            pass
+        colls = [
+            sorted(
+                all_docs_from_collection(rc.client, collname), key=_id_key
+            )
+            for collname in self.needed_dbs
+        ]
+        for db, coll in zip(self.needed_dbs, colls):
+            gtx[db] = coll
+        gtx["all_docs_from_collection"] = all_docs_from_collection
+        gtx["float"] = float
+        gtx["str"] = str
+        gtx["zip"] = zip
+
+    def sout(self):
+        rc = self.rc
+        grants = []
+        if rc.date:
+            desired_date = date_parser.parse(rc.date).date()
+        else:
+            desired_date = dt.date.today()
+        for grant in self.gtx["grants"]:
+            if rc.current and not is_current(grant, now=desired_date):
+                continue
+            grants.append(grant)
+
+        grants_time_info = []
```

Comment:
I think we can remove this whole block if we do something like
```
grants.sort(key=lambda k: get_dates(k).get('end_date'))
```
you may have to futz with it a bit to make it work, but this is cleaner

Reply:
I will need the start and end date eventually when I print the grants to the console. So I could do something like that for sorting, but the current way I have it, it keeps tracks of what the start and end dates are and can use them for printing them when it prints the grants.

Reply:
right, but ``get_dates(grant).get('begin_date')`` gets it.   If you want to make fewer function calls you could do ``grant_dates = get_dates(grant)`` and then you can just grab ``grant_dates.get('begin_date')`` or whateverYour way is ok, but you are recoding conditional logic that is in ``get_dates`` and is fully tested there.  

And while we are on the topic of testing, you have made a new function in ``dates.py`` so it will need tests in ``test_dates.py``.  But it would be better to move it to tools.py and refactor the other is_current(), though this would require us to find everywhere it is imported and change the code calling it, because it has a different signature.  Perhaps it is better to put that on a different PR, but we should do it soon because we now have two functions doing the same thing as each other with the same name in two different modules, and that is a recipe for disaster in thefuture....

Finally, we need a "news".   Please can you ask Songsheng how to do that?  Or I could show people at Group meeting on Friday.

Reply:
For the first part I will look into cleaning it up more. 

I think in regards to your second point, maybe the work we want to do with is_current would be better suited for a different branch. 

I think you should talk about "news" on Friday because it may be helpful to show everyone, especially all of the undergraduates.

Reply:
Also I believe is makes sense for is_current to remain in dates.py because it is a dates method and uses the get_dates method which is also in dates.py (not that it can't call it if it's not in the same file but it may be nice to keep them together)